pom.xml

```diff
@@ -373,14 +373,35 @@
                 <reportingApiConditionalArtifactId>reporting-api-2.0</reportingApiConditionalArtifactId>
             </properties>
         </profile>
-        <profile>
-		    <id>2.4</id>
-		    <properties>
-		        <openMRSVersion>2.4.0</openMRSVersion>
-		        <openMRSMinorVersion>2.4</openMRSMinorVersion>
-		        <jacksonVersion>1.9.13</jacksonVersion>
-		        <reportingApiConditionalArtifactId>reporting-api-2.4</reportingApiConditionalArtifactId>
-		    </properties>
+		<profile>
+			<id>2.3</id>
+			<properties>
+				<openMRSVersion>2.3.0</openMRSVersion>
+				<openMRSMinorVersion>2.3</openMRSMinorVersion>
+				<jacksonVersion>1.9.13</jacksonVersion>
+				<reportingApiConditionalArtifactId>reporting-api-2.0</reportingApiConditionalArtifactId>
+			</properties>
+		</profile>
+		<profile>
+			<id>2.4</id>
+			<modules>
+				<module>api</module>
+				<module>api-1.9</module>
+				<module>api-1.10</module>
+				<module>api-2.0</module>
+				<module>api-2.2</module>
+				<module>api-2.4</module>
+				<module>api-tests</module>
+				<module>api-tests-2.4</module>
+				<module>omod</module>
+			</modules>
```

Comment:
Unfortunately, I think this might be the only way to ensure that api-tests-2.4 is only included when the 2.4 profile is active... Maybe something better could be done here?

Reply:
Personally I think we are well past the point where this module needs to be branched into several maintenance branches for earlier versions, and we don't try to support every OpenMRS release from one module.  Reporting module is pretty stable and there is not a lot changing - most of the changes are to support new core versions and dependency updates.  I'd be interested in your thoughts on this @ibacher .

This is just to say - this is fine, let's do whatever works.  But let's assess how we want to evolve this in the future to better meet today's reporting needs without a lot of the historical baggage.

Reply:
Yeah, that's a very good point. Copying and pasting this block to create a new profile for, e.g. 2.5 and 2.6 will likewise be a bit of a pain. Probably when we have a few more instances running platform 2.4 in production, it would make sense to release reporting-2.0.0 which just supports, e.g. 2.4+ and possibly even removing the 2.4 stuff from the 1.x series. Does that make sense?